src/Symfony/Component/Messenger/Transport/Doctrine/Exception/ConnectionNotFoundException.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Doctrine\Exception;
+
+use Symfony\Component\Messenger\Exception\InvalidArgumentException;
+
+/**
+ * @author Vincent Touzet <vincent.touzet@gmail.com>
+ */
+class ConnectionNotFoundException extends InvalidArgumentException
+{
+    protected $dsn;
+
+    public function __construct(string $dsn = '', int $code = 0, \Throwable $previous = null)
+    {
+        $this->dsn = $dsn;
+        parent::__construct('Unable to retrieve connection for Messenger Doctrine transport', $code, $previous);
+    }
```

Comment:
with a getter on the dsn would be nice. :} And the `$code` I don't think it is useful.